### PR TITLE
special handling of optional fields when destructuring

### PR DESCRIPTION
**Pull request by zth**
Closes https://github.com/rescript-lang/rescript-vscode/issues/714

cc @cknitt

### CODE REVIEW

File: analysis/src/CompletionBackEnd.ml [hunk 1]
```diff
@@ -1047,10 +1047,10 @@ let printConstructorArgs argsLen ~asSnippet =
   if List.length !args > 0 then "(" ^ (!args |> String.concat ", ") ^ ")"
   else ""
 
-type completionMode = Pattern | Expression
+type completionMode = Pattern of Completable.patternMode option | Expression
```
Comment (cknitt) [inline analysis/src/CompletionBackEnd.ml:1050]: I am wondering if we can get rid of the `option` here by having

```ocaml
type patternMode = Default | Destructuring
```

or something like that?

**Review by cristianoc — COMMENTED** [review]
Does this affect destructuring only, or also explicit switches?

**Comment by zth** [comment]
Right now it's configured to only affect let destructures.

**Review by cristianoc — APPROVED** [review]

File: analysis/src/CompletionBackEnd.ml (hunk 1 above)
Comment (cristianoc) [reply analysis/src/CompletionBackEnd.ml:1050]: Good point.

**Pull request merged**